### ISSUE TITLE
no-invalid-interactive cannot be disabled

### ISSUE THREAD

**Issue by mydea** [ember-template-lint/ember-template-lint#889]
When updating to 1.6.2, I run into the following problem:

I have this template:

```hbs
{{! template-lint-disable no-inline-styles }}

{{! We add a transparent circle underneath the actual circle to make it easier clickable }}
<g data-test-chart-line-circle-container>
  <circle
    r="0.8rem"
    cx={{@x}}
    cy={{@y}}
    class="chart-line-circle__padding"
    data-test-chart-line-circle-padding
    {{on "click" this.clickCircle}}
    {{on "mouseenter" this.focusCircle}}
    {{on "mouseleave" this.blurCircle}}
  />

  <circle
    r="0.15rem"
    cx={{@x}}
    cy={{@y}}
    class="chart-line-circle {{if this.isFocused "chart-line-circle--is-focused"}}"
    style={{this.svgPathStyle}}
    data-test-chart-line-circle

    ...attributes
  />
</g>
```

Which now (correctly) raises 

```error  Interaction added to non-interactive element  no-invalid-interactive```

If I try to disable this:

```hbs
{{! template-lint-disable no-inline-styles no-invalid-interactive }}
...
```

I get: 

```
-:-  error  Cannot read property 'length' of undefined  undefined
```

**Comment by rwjblue** [comment]
Hmm, is this only about no-invalid-interactive or _all_ rules?

**Comment by mydea** [comment]
Well, `no-inline-styles` is already disabled in that file and that works fine :)

**Comment by MelSumner** [comment]
I think it's good that it caught this- you'd need some kind of role on the interactive element.

**Comment by rwjblue** [comment]
> I think it's good that it caught this- you'd need some kind of role on the interactive element.

Ya, but that's not exactly what this issue is tracking. A quick recap here is:

* We landing a "breaking bugfix" in `no-invalid-interactive` (https://github.com/ember-template-lint/ember-template-lint/pull/864), to ensure that it checks for `{{on` usage. This is definitely a bugfix (and as such not a "SemVer violation), but it still means @mydea's app can't pass linting.
* There is some sort of failure when attempting to disable that particular rule with the inline comment.

tldr; Disabling with the inline comment should **never** cause this kind of fatal failure, and the fact that it does means we have a logic error.

**Comment by mydea** [comment]
> I think it's good that it caught this- you'd need some kind of role on the interactive element.

Yeah, I know - and actually appreciate the rule very much :) We are in the process of refactoring/changing this to be better usable with a keyboard etc., but in the meanwhile it prevents us from updating ember-template-lint :)

**Comment by Turbo87** [comment]
@mydea I tried to reproduce this with the template you provided and with v1.6.2 the `{{! template-lint-disable no-inline-styles no-invalid-interactive }}` comment seems to work fine for me 🤔 

I also tried this on the v2 beta release and there it works too.

Can you come up with a reproduction in a fresh `ember new` app that you can share?

**Comment by mydea** [comment]
I just tried again on 1.10, and it works now! Not sure if/what has changed though :thinking: Anyhow, thanks for looking into it!
